iNaturalist/src/main/java/org/inaturalist/android/ObservationGridAdapter.java

```diff
@@ -117,16 +118,19 @@ public View getView(int position, View convertView, ViewGroup parent) {
             }
         } else {
             JSONObject observationPhoto;
+            //noinspection TryWithIdenticalCatches
             try {
-                String url;
+                String url = null;
                 observationPhoto = observationPhotos.getJSONObject(0);
 
                 if (isNewApi) {
                     url = observationPhoto.optString("url");
                 } else {
                     JSONObject innerPhoto = observationPhoto.optJSONObject("photo");
-                    url = (innerPhoto.isNull("small_url") ? innerPhoto.optString("original_url") : innerPhoto.optString("small_url"));
-                    if ((url == null) || (url.length() == 0)) url = innerPhoto.optString("url");
+                    if (innerPhoto != null) {
+                        url = (innerPhoto.isNull("small_url") ? innerPhoto.optString("original_url") : innerPhoto.optString("small_url"));
+                        if (url.length() == 0) url = innerPhoto.optString("url");
```

Comment:
Why remove the `url == null` condition?

Reply:
The `optString` method is annotated with `@NonNull`. Since we call `optString` for both the true/false on the tertiary op, the url should never be null. (Side note - linter caught it, I'm definintely not that precise!)